docs/src/content/en/docs/workflows/using-with-agents-and-tools.mdx

```diff
@@ -5,346 +5,119 @@ description: "Steps in Mastra workflows provide a structured way to manage opera
 
 # Agents and Tools
 
-Workflow steps are composable and typically run logic directly within the `execute` function. However, there are cases where calling an agent or tool is more appropriate. This pattern is especially useful when:
+Workflow steps can call agents to leverage LLM reasoning or run custom functions from your codebase. You can either invoke them from within a step's `execute` function or compose them directly as steps using `createStep()`.
 
-- Generating natural language responses from user input using an LLM.
-- Abstracting complex or reusable logic into a dedicated tool.
-- Interacting with third-party APIs in a structured or reusable way.
+## When to use agents and tools
 
-Workflows can use Mastra agents or tools directly as steps, for example: `createStep(testAgent)` or `createStep(testTool)`.
+Use agents in workflow steps when you need to solve open-ended tasks that require reasoning or language generation. Use tools when you need to call APIs, query databases, or run custom functions that perform specific operations.
```

Comment:
I'm not sure if this recommendation makes sense. Personally, I don’t understand why I would call a tool from a step instead of a function, unless I already happened to have a tool. Even then, I might prefer to call a function that the step shares directly. Could you help me (and other readers) understand the reasoning here?